HHS/Head-Start-TTADP
# [TTAHUB-3502] Unwind goals on the RTR and Similarity API

## Description of change

This PR does three things: 
- One goal card per grant number, per role on RTR
- Break up goal similarity suggestions by grant
- Update similarity logic to only recommend similar goal text from the same grant

Note:  Per OHS objective roll up will remain as is for now (text and status).

## How to test

- Create a multi recipient/grant AR and approve.
- Make sure the Goal shows as expected on the RTR, one card per goal.
- Find an existing multi grant goal on the RTR make sure it shows one card per goal (and can be edited accordingly).
- Click into a similarity group make sure that all similar goal suggestions are for the same grant.
- Perform a merge and assert the result is expected based on selected goals to merge.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3502


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
